src/tests/system/tests/test_netgroups.py

```diff
@@ -309,3 +312,112 @@ def test_netgroup__uid_gt_2147483647(client: Client, provider: GenericProvider):
         result = client.tools.getent.group(grpname)
         assert result is not None, f"getent group for group '{grpname}' is empty!"
         assert result.name == grpname, f"Group name '{grpname}' did not match result '{result.name}'!"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.ticket(bz=1576852)
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+def test_netgroup__nss_responder(client: Client, provider: GenericProvider):
+    """
+    :title: SSSD nss responder handles correctly netgroup timeout when backend is offline
+    :setup:
+        1. A user (user-1) and a netgroup (ng-1) are created, and the user is added as a member of the netgroup
+        2. SSSD is restarted with a clean configuration to ensure no stale data is present
+    :steps:
+        1. Depending on the type of provider (AD, IPA, Samba, or LDAP), the script updates the SSSD configuration
+            with an incorrect server URI (e.g., typo.dc.hostname)
+        2. SSSD is restarted again to apply the new configuration
+        3. Checks the status of the SSSD domain
+        4. Capture the process ID (PID) of the sssd_nss process
+        5. Try to retrieve the netgroup information again, expecting it to fail since the SSSD domain is offline
+        6. Verify that the SSSD processes (sssd, sssd_be, sssd_nss, sssd_pam) are still running and that the
+            sssd_nss process ID has not changed, indicating that SSSD has not crashed or restarted unexpectedly
```

Comment:
Why is this step necessary? What is the added value it brings us?

Reply:
@danlavu do you think it's worth checking whether sssd crashed or restarted? From my point of view, as long as sssd is able to resolver the netgroup is fine.